File: src/main/java/com/vaadin/demo/component/slider/SliderMinMaxVisible.java
package com.vaadin.demo.component.slider;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.slider.IntegerSlider;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("slider-min-max-visible")
public class SliderMinMaxVisible extends Div {

    public SliderMinMaxVisible() {
        // tag::snippet[]
        IntegerSlider slider = new IntegerSlider("Temperature", 0, 100);
        slider.setValue(50);
        slider.setMinMaxVisible(true);
        add(slider);
        // end::snippet[]
    }

    public static class Exporter extends DemoExporter<SliderMinMaxVisible> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/slider/slider-min-max-visible.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/slider';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('slider-min-max-visible')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-slider
        label="Temperature"
        min="0"
        max="100"
        value="50"
        min-max-visible
      ></vaadin-slider>
      <!-- end::snippet[] -->
    `;
  }
}
